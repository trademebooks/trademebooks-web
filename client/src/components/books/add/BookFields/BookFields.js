import React, { useEffect } from 'react';
import { MDBRow, MDBCol, MDBContainer, MDBInput, MDBBtn } from 'mdbreact';

import './BookFields.scss';

const BookFields = ({
  book,
  addBook,
  createBook
}) => {

  const onChange = (e) => {
    addBook({ ...book, [e.target.name]: e.target.value });
  }

  useEffect(() => {
    console.log('BookFields', { book });
  }, [book]);

  const postBook = (e) => {
    e.preventDefault();

    console.log({ book })
    createBook(book);
  }
  return (
    <>
      <div className="book-fields-container">
        <MDBContainer>
          <MDBRow>
            <MDBCol md="8" sm="12">
              <form>
                <div className="form-group">
                  <MDBInput label="School" onChange={onChange} name="school" value={book.school} />
                </div>
                <div className="form-group">
                  <MDBInput label="Price" onChange={onChange} name="price" value={book.price} />
                </div>
                <div className="form-group">
                  <MDBInput label="Description" type="textarea" rows="5" onChange={onChange} name="description" value={book.description} />
                </div>
                <div className="form-group">
                  <MDBInput label="Location" onChange={onChange} name="location" value={book.location} />
                </div>
              </form>
            </MDBCol>
          </MDBRow>
          <MDBRow>
            <MDBCol md="4" sm="12">
              <MDBBtn outline className="w-100">
                Preview
              </MDBBtn>
            </MDBCol>
            <MDBCol md="8" sm="12">
              <MDBBtn className="w-100" onClick={postBook}>Post Book</MDBBtn>
            </MDBCol>
          </MDBRow>
        </MDBContainer>
      </div>
    </>
  );
};

export default BookFields;
