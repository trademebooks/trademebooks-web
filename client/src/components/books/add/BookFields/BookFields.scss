.book-fields-container {
  border: solid 2px green;
  margin: 1.5rem 0 1.5rem 0;
}