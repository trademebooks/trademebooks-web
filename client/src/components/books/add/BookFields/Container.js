import { connect } from 'react-redux';
import { addBook, createBook } from '../../../../actions/book';
import BookFields from './BookFields';

const mapStateToProps = (state) => ({
  book: state.book.book
});

const mapDispatchToProps = {
  addBook,
  createBook
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(BookFields);
