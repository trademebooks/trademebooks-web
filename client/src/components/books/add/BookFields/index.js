import Container from './Container'; 

const BookFields = Container;

export default BookFields;

