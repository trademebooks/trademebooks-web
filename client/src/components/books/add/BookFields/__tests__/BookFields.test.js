import React from 'react'
import renderer from 'react-test-renderer'

import BookFields from '../BookFields'

describe('Component: BookFields', () => {
  const props = {
    book: jest.fn(),
    addBook: jest.fn(),
    createBook: jest.fn(),
    editBook: jest.fn(),
    updateBook: jest.fn()
  }

  it('should match the snapshot', () => {
    const bookFieldsSnapshot = renderer
      .create(<BookFields {...props} />)
      .toJSON()

    expect(bookFieldsSnapshot).toMatchSnapshot()
  })
})
