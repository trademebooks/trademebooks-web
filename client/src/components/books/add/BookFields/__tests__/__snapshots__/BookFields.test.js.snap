// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: BookFields should match the snapshot 1`] = `
<div
  className="book-fields-container"
>
  <div
    className="container"
    data-test="container"
  >
    <div
      className="row justify-content-center"
      data-test="row"
    >
      <div
        className="col-sm-12 col-md-8"
        data-test="col"
      >
        <form>
          <div
            className="form-group"
          >
            <div
              className="md-form"
            >
              
              <input
                aria-disabled={false}
                className="form-control"
                data-test="input"
                disabled={false}
                name="price"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onInput={[Function]}
                type="text"
                value=""
              />
              <label
                aria-labelledby=""
                className=""
                data-error=""
                data-success=""
                id=""
                onClick={[Function]}
              >
                Price
              </label>
            </div>
          </div>
          <div
            className="form-group"
          >
            <div
              className="md-form"
            >
              
              <textarea
                aria-disabled={false}
                className="md-textarea form-control"
                data-test="input"
                disabled={false}
                name="description"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onInput={[Function]}
                rows="5"
                value=""
              />
              <label
                aria-labelledby=""
                className=""
                data-error=""
                data-success=""
                id=""
                onClick={[Function]}
              >
                Description
              </label>
            </div>
          </div>
        </form>
      </div>
    </div>
    <div
      className="row"
      data-test="row"
    >
      <div
        className="col-sm-12 col-md-12"
        data-test="col"
      >
        <button
          className="btn-default btn Ripple-parent w-100"
          data-test="button"
          onClick={[Function]}
          onMouseUp={[Function]}
          onTouchStart={[Function]}
          type="button"
        >
          Update Book Listing
          <div
            className="Ripple "
            data-test="waves"
            style={
              Object {
                "height": "0px",
                "left": "0px",
                "top": "0px",
                "width": "0px",
              }
            }
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
